api/src/main/java/org/openmrs/module/fhir2/api/translators/CodingTranslator.java

```diff
@@ -33,5 +41,26 @@ public interface CodingTranslator extends ToFhirTranslator<Concept, Coding>, ToO
 	 * @return the corresponding OpenMRS {@link Concept}
 	 */
 	@Override
-	Concept toOpenmrsType(@Nonnull ICoding coding);
+	public Concept toOpenmrsType(@Nonnull ICoding coding) {
+		if (coding.getCode() != null) {
+			CodeableConcept codeableConcept = new CodeableConcept();
+			codeableConcept.addCoding(new Coding(coding.getSystem(), coding.getCode(), coding.getDisplay()));
+			return conceptTranslator.toOpenmrsType(codeableConcept);
+		}
+		return null;
+	}
+	
+	/**
+	 * @return the coding on the CodeableConcept with the given system, or null if none found.
+	 */
+	protected Coding getCodingForSystem(CodeableConcept codeableConcept, String system) {
```

Comment:
hmm wonder if this can be a public static method since its not really tied to an instance of a CodingTranslator/might be useful from outside of a CodingTranslator as well?

Reply:
Yes, actually. I think adding this as a utility method somewhere would be useful.